website/docs/docs/contributing/testing-a-new-adapter.md

```diff
@@ -0,0 +1,449 @@
+
+## Testing your adapter
+
+The framework presented above is available for use by anyone who installs `dbt-core`, and who wishes to define their own test cases.
+
+Building on top of that foundation, we have also built and made available a [package of reusable adapter test cases](https://github.com/dbt-labs/dbt-core/tree/HEAD/tests/adapter), for creators and maintainers of adapter plugins. These test cases cover basic expected functionality, as well as functionality that frequently requires different implementations across databases.
+
+For the time being, this package is also located within the `dbt-core` repository, but separate from the `dbt-core` Python package.
+
+### Categories of tests
+
+In the course of creating and maintaining your adapter, it's likely that you will end up implementing tests that fall into three broad categories:
+
+1. **Basic tests** that every adapter plugin is expected to pass, defined in `tests.adapter.basic`. Given differences across data platforms, these may require slight modification or reimplementation. Significantly overriding or disabling these tests should be with good reason, since each represents basic functionality that dbt users are accustomed to having. For example, if your adapter does not support incremental models, you should disable the test _and_ note that limitation in documentation, READMEs, and usage guides that accompany your adapter.
+
+2. **Optional tests**, for second-order functionality that is common across plugins, but not required for basic use. Your plugin can opt into these test cases by inheriting existing ones, or reimplementing them with adjustments. More tests will be added as we convert older tests defined on dbt-core and mature plugins to use the standard framework.
+
+3. **Custom tests**, for behavior that is specific to your adapter / data platform. Each data warehouse has its own specialties and idiosyncracies. We encourage you to use the same `pytest`-based framework, utilities, and fixtures to write your own custom tests for functionality that is unique to your adapter.
+
+If you run into an issue with the core framework, or the basic/optional test cases—or if you've written a custom test that you believe would be relevant and useful for other adapter plugin developers—please open an issue or PR in the `dbt-core` repository on GitHub.
+
+## Getting started: basic tests
+
+In this section, we'll walk through the three steps to start running our basic test cases on your adapter plugin:
+
+1. Install dependencies
+2. Set up and configure pytest
+3. Define test cases
+
+### Install dependencies
```

Comment:
Maybe we can come back later and number the steps in these procedures. I'm leaving them as-is for now.